packages/dds/merge-tree/src/test/client.spec.ts

```diff
@@ -207,6 +207,324 @@ describe("TestClient", () => {
+			const tileLabel = "EOP";
+			client.insertMarkerLocal(0, ReferenceType.Tile, {
+				[reservedTileLabelsKey]: [tileLabel],
+				[reservedMarkerIdKey]: "some-id",
+			});
+
+			client.insertTextLocal(0, "abc d");
+
+			client.insertMarkerLocal(0, ReferenceType.Tile, {
+				[reservedTileLabelsKey]: [tileLabel],
+				[reservedMarkerIdKey]: "some-id",
+			});
+
+			client.insertTextLocal(7, "ef");
+			client.insertMarkerLocal(8, ReferenceType.Tile, {
+				[reservedTileLabelsKey]: [tileLabel],
+				[reservedMarkerIdKey]: "some-id",
+			});
+			console.log(client.getText());
+
+			assert.equal(client.getLength(), 10, "length not expected");
+
+			const tile = client.searchForTile(5, tileLabel, true);
+
+			assert(tile, "Returned tile undefined.");
+
+			assert.equal(tile.pos, 0, "Tile with label not at expected position");
+		});
+
+		it("Should be able to find non preceding tile position from client with multiple tile", () => {
+			const tileLabel = "EOP";
+			client.insertMarkerLocal(0, ReferenceType.Tile, {
+				[reservedTileLabelsKey]: [tileLabel],
+				[reservedMarkerIdKey]: "some-id",
+			});
+
+			client.insertTextLocal(0, "abc d");
+
+			client.insertMarkerLocal(0, ReferenceType.Tile, {
+				[reservedTileLabelsKey]: [tileLabel],
+				[reservedMarkerIdKey]: "some-id",
+			});
+
+			client.insertTextLocal(7, "ef");
+			client.insertMarkerLocal(8, ReferenceType.Tile, {
+				[reservedTileLabelsKey]: [tileLabel],
+				[reservedMarkerIdKey]: "some-id",
+			});
+			console.log(client.getText());
+
+			assert.equal(client.getLength(), 10, "length not expected");
+
+			const tile = client.searchForTile(5, tileLabel, false);
+
+			assert(tile, "Returned tile undefined.");
+
+			assert.equal(tile.pos, 6, "Tile with label not at expected position");
+		});
+
+		it("Should be able to find non preceding tile with multiple segments and tiles", () => {
+			const tileLabel = "EOP";
+			Array.from({ length: MaxNodesInBlock * 2 }).forEach((_, i) =>
+				client.insertTextLocal(0, i.toString()),
+			);
+			// pad the string with markers on both ends so we never get undefined solely for convenience of this test
+			for (let i = 0; i <= client.getLength(); i += 3) {
+				client.insertMarkerLocal(i, ReferenceType.Tile, {
+					[reservedTileLabelsKey]: [tileLabel],
+					[reservedMarkerIdKey]: "some-id",
+				});
+			}
+			for (let index = 0; index < client.getLength(); index++) {
+				const tile = client.searchForTile(index, tileLabel, false);
+
+				assert(tile, `Returned tile undefined @ ${index}.`);
+
+				const expected = index % 3 === 0 ? index % 3 : 3 - (index % 3);
+				assert.equal(
+					tile.pos,
+					index + expected,
+					`Tile with label not at expected position: index ${index}, tile.pos ${tile.pos}`,
+				);
+			}
+		});
+
+		it("Should be able to find preceding tile with multiple segments and tiles", () => {
+			const tileLabel = "EOP";
+			Array.from({ length: MaxNodesInBlock * 2 }).forEach((_, i) =>
```

Comment:
> To do this, you could add markers/reference positions randomly (with a seed) and validate some random queries against forwardExcursion/backwardExcursion.

I thought about doing this, but I'm not sure I can use forward and backwardExcursion as a comparison anymore. Both call depthFirstNodeWalk on the next (or previos) child, and since searchForTile includes the start position in the search, the comparisons would only be correct if the search position is not a tile.

Reply:
makes sense--in that case I'd tweak the above to write a "slow" variant of searchForTile which first manually checks the first segment and then uses forwardExcursion/backwardExcursion. The idea of having a 'slow' reference comparison implementation which we have higher confidence for correctness is generally a pretty powerful one for increasing test coverage. We use this idea in `testClient.ts`'s `findReconnectionPosition` for example.